pandas/tests/copy_view/test_methods.py

```diff
@@ -575,6 +576,48 @@ def test_sort_index(using_copy_on_write):
     tm.assert_series_equal(ser, ser_orig)
 
 
+@pytest.mark.parametrize(
+    "obj, kwargs", [(Series([1, 2, 3]), {}), (DataFrame({"a": [1, 2, 3]}), {"by": "a"})]
+)
+def test_sort_values(using_copy_on_write, obj, kwargs):
+    obj_orig = obj.copy()
+    obj2 = obj.sort_values(**kwargs)
+
+    if using_copy_on_write:
+        assert np.shares_memory(obj.values, obj2.values)
+    else:
+        assert not np.shares_memory(obj.values, obj2.values)
+
+    # mutating df triggers a copy-on-write for the column / block
+    obj2.iloc[0] = 0
+    assert not np.shares_memory(obj2.values, obj.values)
+    tm.assert_equal(obj, obj_orig)
+
+
+@pytest.mark.parametrize(
+    "obj, kwargs", [(Series([1, 2, 3]), {}), (DataFrame({"a": [1, 2, 3]}), {"by": "a"})]
+)
+def test_sort_values_inplace(using_copy_on_write, obj, kwargs, using_array_manager):
+    obj_orig = obj.copy()
+    view = obj[:]
+    obj.sort_values(inplace=True, **kwargs)
+
+    assert np.shares_memory(obj.values, view.values)
+
+    # mutating obj triggers a copy-on-write for the column / block
+    obj.iloc[0] = 0
+    if (
+        using_copy_on_write
+        or using_array_manager
+        and isinstance(obj, DataFrame)
+        and pd.options.mode.copy_on_write
```

Comment:
ArrayManager never uses Copy-on-Write (we should maybe error when setting those two options at the same time), so this check (and the `or using_array_manager` two lines above) shouldn't be needed. I suppose the reason it seemingly did give a copy is because of the `.values` giving a copy